src/Audio.md

```diff
@@ -0,0 +1,103 @@
+Channel 4 is the "[noise](https://en.wikipedia.org/wiki/Noise_in_music)" channel, producing a pseudo-random wave.
+
+The VIN channel is a signal received directly from the cartridge, allowing external hardware to supply a fifth sound channel.
+No licensed games used this feature, and it was omitted from the Game Boy Advance.
+
+::: tip POCKET MUSIC
+
+Despite rumors, *Pocket Music* does not use VIN. It refuses to run on the GBA for a different reason:
+the developer couldn't figure out how to silence buzzing associated with sample playback on the wave channel.
+
+:::
+
+[^speaker_mono]:
+The speaker merges back the two channels, losing the stereo aspect entirely.
+
+## Common concepts
+
+### APU
+
+The Game Boy's sound chip is called the <abbr title="Audio Processing Unit">APU</abbr>.
+
+The APU runs off the same master clock as the rest of the Game Boy, which is to say, it is fully synced with the CPU and [PPU](<#Rendering Overview>).
+This also means that the APU runs about 2.4% faster on the SGB1, increasing frequencies by as much and thus sounding slightly higher-pitched.
+The SGB2 does not suffer from this issue.
+
+The APU processes a lot of events at a fixed rate, which depends on the type of event.
+Thus, the APU interfaces use durations instead of frequencies, which may be confusing as signal theory and music are more typically based on the latter.
+Thus, durations will be expressed from their frequencies: for example, a "256 Hz tick" means "1/256th of a second".
+
+The length of ticks is not affected by [CGB double speed](<#FF4D - KEY1 - CGB Mode Only - Prepare Speed Switch>), so the APU works just the same regardless of CPU speed.
+
+::: warning
+
+The Game Boy's APU is actually full of tricky details; this chapter will mostly describe the intended / common behavior, and often paper over bugs & quirks.
+Readers wishing to learn more should read the [APU details](<#APU details>) chapter.
+
+:::
+
+### Envelope
+
+Each channel's [**envelope**](https://en.wikipedia.org/wiki/Envelope_(music)) can be configured.
+A signal's [envelope](https://en.wikipedia.org/wiki/Envelope_(waves)) is, roughly speaking, the variation of the signal's amplitude.
+On the GB, "envelope" basically designates the channel's loudness.
+
+All channels can have their envelope controlled to modulate their volume (CH3 less precisely than other channels).
+In addition, all channels but CH3 benefit from a functionality called **envelope sweep**, which automatically adjusts the envelope over time.
+The parameters that can be controlled are how many times the envelope will be stepped, in which direction, and how long between steps.
+Envelope sweep ticks at 64 Hz, and every 1&ndash;7 of those ticks, the envelope will be increased or decreased.
+
+Unfortunately, of [ADSR envelopes](https://en.wikipedia.org/wiki/ADSR_envelope), the Game Boy only provides for Attack and Sustain, via envelope sweep and sound length respectively.
```

Comment:
This is not really correct. In envelope terms, the GB envelope generators would be either attack/sustain or just decay, depending on the direction. And while the sound length register could be used to emulate (effectively) sustain and instant release, it could also be used to stop the channel during the decay phase. I don't think it's necessarily a good idea to try to fit the GB envelope control into exact ADSR terminology, but rather try to explain what each function does on its own.

Reply:
Okay, if you say so.